File: Problem Set 5/GokuAndDiscount.java
import java.io.*;
import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
       
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Scanner sc = new Scanner(System.in);
        // Write your code here
        int N = Integer.parseInt(br.readLine());
        String s[] = br.readLine().split(" ");
        int ar[] = new int[N+7];
        for(int i=0; i<N; i++){
            ar[i] = Integer.parseInt(s[i]);
        }
        Arrays.sort(ar,0,N);
        long sum = 0, count=0;
        for(int i=0; i<N; i++){
            if(ar[i]>=sum){
                sum += ar[i];
            }
            else count++;
        }
        System.out.println(N-count);
    }
}

File: Problem Set 5/goku_and_discount.cpp
/*
// Sample code to perform I/O:

#include <iostream>

using namespace std;

int main() {
	int num;
	cin >> num;										// Reading input from STDIN
	cout << "Input number is " << num << endl;		// Writing output to STDOUT
}

// Warning: Printing unwanted or ill-formatted data to output will cause the test cases to fail
*/

// Write your code here
#include <bits/stdc++.h>
using namespace std;
#define ll long long 
#define mod 1000000009
typedef pair<int,int> pp;
#define ff first 
#define ss second 
#define pb push_back
#define lb lower_bound 
#define ub upper_bound 
int main(){
	int n,ans=0;
	cin>>n;
	ll a[n];
	for(int i=0;i<n;i++) cin>>a[i];
	sort(a,a+n);
	ll sum=0;
	for(int i=0;i<n;i++){
		if(sum<=a[i]){
			ans++;
			sum+=a[i]; 
		}
	}
	cout<<ans;
}